src/plugins/tool_drag.py

```diff
@@ -85,7 +155,46 @@ def on_drag(self, target):
 
         if ty != 0 or tx != 0 or tz != 0:
             self._mouse = (target.mouse_x, target.mouse_y)
+        if len(target.voxels._selection) == 0:
+            if self.fixeddirection:
+                delt = tx+ty+tz
+                if self.xdir:
+                    target.voxels.translate(delt, 0, 0)
+                if self.ydir:
+                    target.voxels.translate(0, delt, 0)
+                if self.zdir:
+                    target.voxels.translate(0, 0, delt)
+            else:
+                target.voxels.translate(tx, ty, tz)
+        else:
+            if self.fixeddirection:
+                delt = tx + ty + tz
+                if delt != 0:
+                    self.pastoffset += delt
```

Comment:
The bug is caused by the last 3 lines. It always happen when locking a direction and dragging diagonal: 
when dragging from...
* top-right to bottom-left or vice versa: delt=2 resulting in alternating filled and empty stamps
* top-left to bottom-right or vice versa: delt=0 nothing happens, which is most likely also not what the user wants